framework/src/dirackernels/FunctionDiracSource.C

```diff
@@ -0,0 +1,57 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "FunctionDiracSource.h"
+#include "Function.h"
+
+template <>
+InputParameters
+validParams<FunctionDiracSource>()
+{
+  InputParameters params = validParams<DiracKernel>();
+  params.addRequiredParam<FunctionName>(
+      "function", "The function to use for controlling the specified dirac source.");
+  params.addRequiredParam<std::vector<Real>>("point", "The x,y,z coordinates of the point");
+  return params;
+}
+
+FunctionDiracSource::FunctionDiracSource(const InputParameters & parameters)
+  : DiracKernel(parameters),
+    _function(getFunction("function")),
+    _point_param(getParam<std::vector<Real>>("point"))
```

Comment:
This is probably copied from `ConstantPointSource` but it's really the old way of reading in Points. Just use `getParam<Point>` instead.